packages/ember-data/lib/system/model/model.js

```diff
@@ -33,6 +33,11 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   store: null,
 
+  // Remove support for `record.container`
+  get container() {
+    Ember.assert("A model's private reference to the container has been removed for consistency reasons. Instances of a store (i.e. `this.store`) have access to the container, although usage from within a model is discouraged and can lead to unintended consequences.");
+  },
+
```

Comment:
I'm in favor of removing. Telling people that we moved a private API and encouraging them to use another private API seems bad.